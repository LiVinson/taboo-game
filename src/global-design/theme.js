const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
