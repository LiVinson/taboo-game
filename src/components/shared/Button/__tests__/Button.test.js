import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import "jest-styled-components"
import { ThemeProvider } from "styled-components"
import theme from "../../../../global-design/theme"
import { Button, PrimaryButton } from "../index.js"

test("Button renders correctly", () => {
  const props = {
    text: "Button",
    onClick: jest.fn()
 
  }
  const wrapper = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button {...props} />
      </ThemeProvider>
    )
    .toJSON()
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toHaveStyleRule("background-color", theme.color.offWhite)
})

test("Primary button renders correctly", () => {
    const props = {
      text: "Button",
      onClick: jest.fn()   
    }

    const wrapper = renderer
      .create(
        <ThemeProvider theme={theme}>
          <PrimaryButton {...props} />
        </ThemeProvider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).not.toHaveStyleRule("background-color", theme.color.offWhite)

  })

/*
    receives a text prop and a onClick property
*/
test("Button displays text received as prop", () => {
  const props = {
    text: "A button",
    onClick: jest.fn(),
  }
  const wrapper = mount(<Button {...props} theme={theme} />)
  expect(wrapper.find("button").text()).toEqual("A button")
})

test("Button calls onClick prop when clicked", () => {
  const props = {
    text: "A button",
    onClick: jest.fn(),
  }
  const wrapper = mount(<Button {...props} theme={theme}/>)
  wrapper.find("button").simulate("click")
  expect(props.onClick.mock.calls.length).toBe(1)
})
