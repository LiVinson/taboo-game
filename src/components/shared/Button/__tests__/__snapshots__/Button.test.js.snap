// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c0:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c0:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c0:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  Button
</button>
`;

exports[`Primary button renders correctly 1`] = `
.c0 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c0:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c0:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c0:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c0:focus {
  outline: solid 2px #f7f9f9;
}

<button
  className="sc-AxjAm c0"
  onClick={[MockFunction]}
>
  Button
</button>
`;
